mindsdb/integrations/handlers/langchain_handler/langchain_handler.py

```diff
@@ -212,8 +212,21 @@ def create_agent(self, df: pd.DataFrame, args: Dict=None, pred_args: Dict=None)
 
         # Set up embeddings model if needed.
         if args.get('mode') == 'retrieval':
-            # get args for embeddings model
+
             embeddings_args = args.pop('embedding_model_args', {})
+
+            # no embedding model args provided, use same provider as llm
+            if not embeddings_args:
+                logger.warning("'embedding_model_args' not found in input params, "
+                               "Trying to use the same provider and model as used for llm. "
+                               f"provider: {args['provider']}"
+                               f"model: {args['model_name']}"
+                               )
+
+                # get args for embeddings model
+                embeddings_args['class'] = args['provider']
+                embeddings_args['model'] = args['model_name']
```

Comment:
will the main llm model return embeddings (not text response)?

Reply:
It's only used to construct embedding model class of langchain. If model is not supported for embeddings it will throw an error

Reply:
Actually ill remove model, that way it will use default model for embedding provider